core/network-libp2p/src/transport.rs

```diff
@@ -34,39 +36,81 @@ pub fn build_transport(
 	keypair: identity::Keypair,
 	wasm_external_transport: Option<wasm_ext::ExtTransport>
 ) -> (Boxed<(PeerId, StreamMuxerBox), io::Error>, Arc<bandwidth::BandwidthSinks>) {
+	// Build configuration objects for encryption mechanisms.
+	#[cfg(not(target_os = "unknown"))]
+	let noise_config = {
+		let noise_keypair = noise::Keypair::new().into_authentic(&keypair)
+			// For more information about this panic, see in "On the Importance of Checking
+			// Cryptographic Protocols for Faults" by Dan Boneh, Richard A. DeMillo,
+			// and Richard J. Lipton.
+			.expect("can only fail in case of a hardware bug; since this signing is performed only \
```

Comment:
I don’t think we need to go into all this detail here, unless the hardware failure could be in an external HSM or similar.  Unknown hardware bugs in the main system (CPU/memory) that cause an incorrect computation are Undefined Behavior, full stop ― they could just as easily corrupt the kernel’s page tables or something else that is equally fatal.  I think that everything after the `;` could be replaced with `hardware bugs are fatal; qed`.